File: core/src/main/java/inetsoft/web/viewsheet/event/MoveSelectionChildEvent.java
/*
 * inetsoft-core - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package inetsoft.web.viewsheet.event;


/**
 * Class that encapsulates the parameters for moving a selection child.
 *
 * @since 12.3
 */
public class MoveSelectionChildEvent {
   public int getFromIndex() {
      return fromIndex;
   }

   public void setFromIndex(int fromIndex) {
      this.fromIndex = fromIndex;
   }

   public int getToIndex() {
      return toIndex;
   }

   public void setToIndex(int toIndex) {
      this.toIndex = toIndex;
   }

   public boolean isCurrentSelection() {
      return currentSelection;
   }

   public void setCurrentSelection(boolean currentSelection) {
      this.currentSelection = currentSelection;
   }

   @Override
   public String toString() {
      return "{fromIndex: " + fromIndex +
         ", toIndex: " + toIndex +
         ", currentSelection: " + currentSelection +
         "}";
   }

   private int fromIndex;
   private int toIndex;
   private boolean currentSelection;
}

File: web/projects/portal/src/app/vsobjects/event/move-selection-child-event.ts
/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
import { ViewsheetEvent } from "../../common/viewsheet-client/index";

/**
 * Class that encapsulates an event sent to the server to instruct it to move a
 * selection child in a selection container.
 */
export class MoveSelectionChildEvent implements ViewsheetEvent {
   public fromIndex: number;
   public toIndex: number;
   public currentSelection: boolean;

   /**
    * Creates a new instance of <tt>MoveSelectionChildEvent</tt>.
    *
    * @param fromIndex           the index to move from.
    * @param toIndex             the index to move to.
    * @param currentSelection    if it is a current selection.
    */
   constructor(fromIndex: number, toIndex: number, currentSelection: boolean)
   {
      this.fromIndex = fromIndex;
      this.toIndex = toIndex;
      this.currentSelection = currentSelection;
   }
}
